accepted/2020/platform-exclusion/platform-exclusion.md

```diff
@@ -0,0 +1,217 @@
+* Building a class library
+    - Building a regular class library (targeting `netstandard` or `net5.0`)
+      will not complain about APIs that are unsupported in Blazor.
+* Building a class library for use in Blazor Web Assembly
+    - Developer can modify the project file to explicitly indicate that they
+      want to see issues with APIs that are unsupported in Blazor.
+
+## Requirements
+
+### Goals
+
+* Be able to mark APIs that are unsupported on a particular OS
+* Allows platforms to be considered supported by default with the ability to add
+  or remove platforms from the project file and SDK.
+* Blazor Web Assembly is only considered supported when building a Blazor Web
+  Assembly app. For regular class libraries (`netstandard` or `net5.0`) the
+  default assumes that it's not supported and thus won't generate any warnings.
+  However, the developer must still be able to manually include Blazor Web
+  Assembly manually.
+* The annotations for unsupported APIs can express that an API was only
+  unsupported for a specific version.
+
+### Non-Goals
+
+## Design
+
+### Attribute
+
+```C#
+namespace System.Runtime.Versioning
+{
+    // Rename RemovedInOSPlatformAttribute to
+    // UnsupportedOSPlatformAttribute
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple=true, Inherited=false)]
+    public sealed class UnsupportedOSPlatformAttribute : OSPlatformAttribute
+    {
+        public UnsupportedOSPlatformAttribute(string platformName);
+    }
+```
+
+The semantics of this attribute are as follows:
+
+* Marks an assembly, module, or API as unsupported on a given OS platform.
+* The `platformName` can contain no version number, a single version number, or
+  a range:
+    - `browser`. Marks the API as unsupported on all version of `browser`. This
+      will be used for APIs that can't be supported on a particular OS platform.
+    - `ios14.0`. Marks the API as unsupported starting with iOS 14.0. This will
+      be used for OS platform APIs that were removed/disabled.
+    - `windows0.0-10.0.19041`. Marks the API as unsupported for 10.0.19041 and
+      earlier. This will be used for APIs that were unsupported before but can
+      be supported starting with a later version of the OS. This is different
+      from applying `[MinimumOSPlatform("windows10.0.19041")]` in that the
+      presence of this attribute doesn't cause the API to be considered
+      unsupported on other platforms.
+
+### Analyzer behavior
+
+The existing analyzer that handles `[MinimumOSPlatform]` will be modified to
+handle `[UnsupportedOSPlatform]` (previously named `[RemovedInOSPlatform]`)
```

Comment:
[platform-checks.md](https://github.com/dotnet/designs/blob/0c0581895fc59ac03c02088b4407cf73d3562943/accepted/2020/platform-checks/platform-checks.md) should be updated to mention the renaming of this attribute.

Reply:
Will do if this is the direction we're going with.